### PR TITLE
Switch to using fields.py and align with upstream development

### PR DESCRIPTION
**Pull request by roelof-groenewald**
The biggest change in this PR is to switch to using `fields.py` to handle full domain `rho` and `phi` arrays in Python. This change caused several unit test references to change since the old method we used set ghost-cell values to 0 whereas the `fields.py` approach has non-zero values in ghost cells. There also seems to be a slight difference in the final values of the `test_poisson_pseudo_1d.py` test as shown in the figure below:
phi_comparison

Other changes are meant to bring our fork closer into alignment with upstream `development`. Part of this clean-up involved deleting duplicate functions. Currently our fork differs from upstream in the following files (excluding files in the `mewarpx` directory):
```
 .circleci/config.yml                                                       |  144 ++++
 .circleci/export_vars.sh                                                   |   10 +
 .readthedocs.yml                                                           |   38 +-
 Python/pywarpx/Bucket.py                                                   |    6 +
 Python/pywarpx/_libwarpx.py                                                |    3 +
 Source/FieldSolver/ElectrostaticSolver.cpp                                 |   22 +
 Source/Make.WarpX                                                          |    2 +-
 Source/Python/WarpXWrappers.H                                              |   10 +
 Source/Python/WarpXWrappers.cpp                                            |  117 +++
 Source/WarpX.H                                                             |    3 +
 buildspec.yml                                                              |   63 ++
```

### CODE REVIEW

File: mewarpx/mewarpx/mwxrun.py [hunk 1]
```diff
@@ -285,10 +290,10 @@ def get_gathered_rho_grid(self, species_name=None, include_ghosts=True):
             _libwarpx.libwarpx.warpx_depositRhoSpecies(
                 ctypes.c_char_p(species_name.encode('utf-8'))
             )
-        return _libwarpx._get_mesh_field_list(
-            _libwarpx.libwarpx.warpx_getGatheredChargeDensityFP,
-            self.lev, None, include_ghosts
-        )
+        if include_ghosts:
+            return self.rho_wrapper_ghosts[Ellipsis]
```
Comment (KZhu-ME) [inline mewarpx/mewarpx/mwxrun.py:294]: What's the difference between returning `self.rho_wrapper_ghosts[Ellipsis]` and `self.rho_wrapper_ghosts`, won't both just return the full array?

Comment (KZhu-ME) [reply mewarpx/mewarpx/mwxrun.py:294]: Nevermind, after looking through fields.py, the multifab wrapper object that is returned has specific behavior with Ellipsis that converts it to a numpy array.

**Review by KZhu-ME — APPROVED** [review]
Looks good!

**Review by EtherealMC-Bit — APPROVED** [review]

**Review by peterscherpelz — APPROVED** [review]
I don't see any issues. Feel free to merge once your own tests are complete.

File: mewarpx/mewarpx/poisson_pseudo_1d.py [hunk 2]
```diff
@@ -133,11 +132,9 @@ def _run_solve(self):
...
 
         # get rho from WarpX
-        self.rho_data = mwxrun.get_gathered_rho_grid()
-        # run superLU solver only on the root processor
-        if mwxrun.me == 0:
-            self.rho_data = self.rho_data[0][:,:,0]
-            self.solve()
+        self.rho_data = mwxrun.get_gathered_rho_grid()[:,:,0]
+        # run superLU solver to get phi
+        self.solve()
```
Comment (PhilMiller) [inline mewarpx/mewarpx/poisson_pseudo_1d.py:137]: There's probably some performance tradeoff here of replicated vs centralized solve calls. The direct solver is such a tiny fraction of execution in the runs we've seen that it probably will never matter, though.

File: mewarpx/tests/test_assemblies.py [hunk 3]
```diff
@@ -157,8 +157,8 @@ def test_embedded_rectangle():
     # Check phi results against reference data                            #
     #######################################################################
 
-    phi = mwxrun.get_gathered_phi_grid()[0]
-
+    phi = mwxrun.get_gathered_phi_grid(include_ghosts=False)
+    # np.save('embedded_rectangle_phi.npy', phi)
```
Comment (PhilMiller) [inline mewarpx/tests/test_assemblies.py:161]: What's this commented line doing here?

Comment (roelof-groenewald) [reply mewarpx/tests/test_assemblies.py:161]: It is there in case we want to regenerate the reference data.

**Pull request merged**